OpenRA.Mods.Common/Widgets/Logic/Ingame/GameInfoStatsLogic.cs

```diff
@@ -53,9 +54,17 @@ public GameInfoStatsLogic(Widget widget, World world)
 				nameLabel.GetColor = () => pp.Color.RGB;
 
 				var flag = item.Get<ImageWidget>("FACTIONFLAG");
-				flag.GetImageName = () => pp.Country.Race;
 				flag.GetImageCollection = () => "flags";
-				item.Get<LabelWidget>("FACTION").GetText = () => pp.Country.Name;
+				if (pp.World.RenderPlayer != null && pp.World.RenderPlayer.Stances[pp] == Stance.Ally)
```

Comment:
[Observers don't see that panel at all](https://github.com/OpenRA/OpenRA/blob/74c22523aa11b9b6254e5593a69d24199ecb7f7c/OpenRA.Mods.Common/Widgets/Logic/Ingame/GameInfoLogic.cs#L32), so a null check for the local player would be kind of useless. But I agree that the factions should be revealed at the end of a game, so I suggest
```csharp
if ((pp.World.RenderPlayer != null && pp.World.RenderPlayer.Stances[pp] == Stance.Ally) || lp.WinState != WinState.Undefined)
```